c7n/utils.py

```diff
@@ -375,3 +376,14 @@ def reformat_schema(model):
             ret[key]['required'] = True
 
     return ret
+
+
+_profile_session = None
+def get_profile_session(options):
```

Comment:
always a bit wary of globals.. effectively we have two sessions that are being used for purposes other than connecting to the aws api, one to get the sdk's notion of a default region, and one to have it parse the sdk json files, just retaining two session instantiations seems better than deal with a cached global that doesn't need valid credentials for its extant uses.